server/instances.go

```diff
@@ -151,6 +151,29 @@ func (p *Plugin) InstallInstance(instance Instance) error {
 				}
 			}
 
+			if instance.Common().Type == CloudOAuthInstanceType && len(instances.IDs()) > 0 && instances.checkIfExists(instance.GetID()) {
```

Comment:
I originally thought we also want to check if the previous instance is `cloud` in this if condition, but I see in a block below we have a special case for when the previous instance was `cloud-oauth` as well